pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -235,6 +236,52 @@ def brick_sort_parallel(array, num_threads, **kwargs):
     if _check_type(array, DynamicArray):
         array._modify(force=True)
 
+def heapsort(array, **kwargs):
+    """
+    Implements Heap Sort sorting algorithm
+
+    Parameters
+    ==========
+
+    array: Array
+        The array which is to be sorted.
+    start: int
+        The starting index of the portion
+        which is to be sorted.
+        Optional, by default 0
+    end: int
+        The ending index of the portion which
+        is to be sorted.
+        Optional, by default the index
+        of the last position filled.
+
+    Examples
+    ========
+    >>> from pydatastructs import OneDimensionalArray, heapsort
+    >>> arr = OneDimensionalArray(int,[3, 2, 1])
+    >>> heapsort(arr)
+    >>> [arr[0], arr[1], arr[2]]
+    [1, 2, 3]
+    """
+    from pydatastructs.trees.heaps import BinaryHeap
+
+    start = kwargs.get('start', 0)
+    end = kwargs.get('end', len(array) - 1)
+
+    h = BinaryHeap(heap_property="min")
+    for i in range(start, end+1):
+        if array[i] is not None:
+            h.insert(array[i], array[i])
```

Comment:
```suggestion
            h.insert(array[i])
```

Reply:
insert takes 2 arguments, key and data

Reply:
`def insert(self, key, data):`

Reply:
`data` should be optional for consistency, at all places, some of them are done, may be in a different patch. `key` should be required.

Reply:
should I create another PR for that or push it in here?

Reply:
> should I create another PR for that or push it in here?

Create a separate PR and make data optional wherever, `key, data` combination is passed as an argument.